src/applications/auth/containers/AuthApp.jsx

```diff
@@ -6,15 +6,88 @@ import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
 import siteName from '../../../platform/brand-consolidation/site-name';
+import recordEvent from '../../../platform/monitoring/record-event';
 import { toggleLoginModal } from '../../../platform/site-wide/user-nav/actions';
 import { authnSettings } from '../../../platform/user/authentication/utilities';
 import {
   hasSession,
   setupProfileSession,
 } from '../../../platform/user/profile/utilities';
 import { apiRequest } from '../../../platform/utilities/api';
+import get from '../../../platform/utilities/data/get';
 
-import recordEvent from '../../../platform/monitoring/record-event';
+class AuthMetrics {
+  constructor(type, payload) {
+    this.type = type;
+    this.payload = payload;
+    this.userProfile = get('data.attributes.profile', payload, {});
+    this.loaCurrent = get('loa.loaCurrent', this.userProfile, null);
+    this.serviceName = get('signIn.serviceName', this.userProfile, null);
+  }
+
+  compareLoginPolicy = () => {
+    // This is experimental code related to GA that will let us determine
+    // if the backend is returning an accurate service_name
+
+    const attemptedLoginPolicy =
+      this.type === 'mhv' ? 'myhealthevet' : this.type;
+
+    if (this.serviceName !== attemptedLoginPolicy) {
+      recordEvent({
+        event: `login-mismatch-${attemptedLoginPolicy}-${this.serviceName}`,
+      });
+    }
+  };
+
+  recordGAAuthEvents = () => {
+    switch (this.type) {
+      case 'signup':
+        recordEvent({ event: `register-success-${this.serviceName}` });
+        break;
+      case 'mhv':
+      case 'dslogon':
+      case 'idme':
+        recordEvent({ event: `login-success-${this.serviceName}` });
+        this.compareLoginPolicy();
+        break;
+      /*
+      case 'mfa':
+        recordEvent({ event: `multifactor-success-${this.serviceName}` });
+        break;
+      case 'verify':
+        recordEvent({ event: `verify-success-${this.serviceName}` });
+        break;
+      */
```

Comment:
Still need to discuss the naming for these events. In the meantime, will retain the old `login-or-register-success` event below.